src/main/webapp/app/course/competencies/create/create-course-competency.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit } from '@angular/core';
+import { onError } from 'app/shared/util/global.utils';
+import { ActivatedRoute, Router } from '@angular/router';
+import { AlertService } from 'app/core/util/alert.service';
+import { finalize, switchMap, take } from 'rxjs/operators';
+import { HttpErrorResponse } from '@angular/common/http';
+import { LectureService } from 'app/lecture/lecture.service';
+import { Lecture } from 'app/entities/lecture.model';
+import { DocumentationType } from 'app/shared/components/documentation-button/documentation-button.component';
+
+@Component({ template: '' })
+export abstract class CreateCourseCompetencyComponent implements OnInit {
+    readonly documentationType: DocumentationType = 'Competencies';
+
+    isLoading: boolean;
+    courseId: number;
+    lecturesWithLectureUnits: Lecture[] = [];
+
+    constructor(
+        protected activatedRoute: ActivatedRoute,
+        protected router: Router,
+        protected alertService: AlertService,
+        protected lectureService: LectureService,
```

Comment:
you sometimes use inject() and sometimes constructor injection is there a reason for it? If you use inject here the child classes do not have to pass in the services via the super constructor.

Reply:
I did not really create new components but only refactored and moved existing ones. So if that particular component for the competency used constructor injection before it still does so now.